taipy/core/data/_file_datanode_mixin.py

```diff
@@ -100,7 +101,15 @@ def _get_downloadable_path(self) -> str:
 
         return ""
 
-    def _upload(self, path: str, upload_checker: Optional[Callable[[str, Any], bool]] = None) -> ReasonCollection:
+    def _upload(
+        self,
+        data: Optional[Any] = None,
+        path: str = "",
+        upload_checker: Optional[Callable[[str, Any], bool]] = None,
+        job_id: Optional["JobId"] = None,
+        editor_id: Optional[str] = None,
+        **kwargs: Dict[str, Any]
+    ) -> None:
```

Comment:
I don't get this change. 

You only need one new argument: an optional `editor_id`. In particular, we don't want data or job_id. We must also return a reason collection!

Why not add a kwargs parameter to pass it to the track edit method, indeed. But this is out of the scope of the issue.

```suggestion
    def _upload(self, path: str, upload_checker: Optional[Callable[[str, Any], bool]] = None, **kwargs: Dict[str, Any]) -> ReasonCollection:
```